### PR TITLE
fix(Select): close menu on second click

### PR DESCRIPTION
**Pull request by teimurjan**
[FX-965](https://toptal-core.atlassian.net/browse/FX-965)

### Description

Fixed the bug when you click on open Select and it doesn't close.

#### Default

1. Go to `Select` chapter.
2. Click on select in the default example.
3. Click on select again.
4. Be sure that the menu is closed.

#### Multiple

1. Go to `Select` chapter.
2. Click on select in the multiple example.
3. Choose an option.
4. Click on select again.
5. Be sure that the menu is closed.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Review by vedrani — APPROVED** [review]

File: packages/picasso/src/Select/hooks/use-select-props/use-click-handler/test.ts [hunk 1]
```diff
@@ -15,10 +15,23 @@ describe('useClickHandler', () => {
...
     props.selectState.canOpen = false
+    props.selectState.isOpen = true
+    const { result } = renderHook(() => useClickHandler(props))
+
+    result.current()
+
+    expect(props.selectState.close).toHaveBeenCalledTimes(1)
+  })
+
+  it("does nothing when can't be open and not open", () => {
```
Comment (havenchyk) [inline packages/picasso/src/Select/hooks/use-select-props/use-click-handler/test.ts:30]: this name confuses me, could you rephrase?

File: packages/picasso/src/NonNativeSelect/test.tsx [hunk 2]
```diff
@@ -286,6 +285,21 @@ describe('NonNativeSelect', () => {
     expect(queryByRole('menu')).not.toBeInTheDocument()
   })
 
+  it('closes menu when open and select is clicked', () => {
```
[the remark this answers is not in the document]
Comment (teimurjan) [reply packages/picasso/src/NonNativeSelect/test.tsx]: ✅

File: packages/picasso/src/Select/hooks/use-select-props/use-click-handler/test.ts (hunk 1 above)
Comment (teimurjan) [reply packages/picasso/src/Select/hooks/use-select-props/use-click-handler/test.ts:30]: Agreed on not changing this place and discuss inner hooks testing within TF.

**Review by havenchyk — APPROVED** [review]
